traceevents/src/main/java/com/tomtom/kotlin/traceevents/TraceDiagnosticContext.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2012-2021, TomTom (http://tomtom.com).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tomtom.kotlin.traceevents
+
+/**
+ * This class allows storing ThreadLocal trace context.
+ *
+ * Inspired by basic MDC implementation from org.org.slf4j.helpers.BasicMDCAdapter.
+ * This implementation allows storing any object type, not just strings, in a map.
+ */
+object TraceDiagnosticContext {
```

Comment:
This is probably because I'm not used to MDC, but to me `TraceDiagnosticContext` doesn't sound like the most intuitive name for storing thread local context.

I would say that something like `TraceThreadLocalContext` would be easier to understand for code readers as well as discovering this functionality in the library...

Reply:
Yes and no. I thought about it and left the name to include `...DiagnosticContext` is it so closely resembles MDC. Perhaps you're right and we should just mention the MDC pattern clearly, but have a more descriptive name.